client/signup/steps/design-type/index.jsx

```diff
@@ -28,6 +28,14 @@ class DesignTypeStep extends Component {
 		translate: identity
 	};
 
+	choiceHandlers = {};
+
+	componentWillMount() {
+		this.choiceHandlers = transform( this.getChoices(), ( handlers, choice ) => {
+			handlers[ choice.type ] = ( event ) => this.handleChoiceClick( event, choice.type );
+		}, {} );
+	}
```

Comment:
This looks good. An alternative impl occurred to me that eschews `componentWillMount`:

```js
class DesignTypeStep { …
  getChoiceHandlers = memoize( () =>
    transform( this.getChoices(), …, {} ) );
}
```

leaving you free to call it in `renderChoices`. Don't go and implement it, just wanted to share the thought. :)